### PR TITLE
Add code scanning workflow

### LINKED ISSUE #2607: Add code scanning workflow to audit, at least, npm signatures

### PR DESCRIPTION
**Pull request by kevinszuchet**
Closes #2607

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: "CodeQL"
+
+on:
+  pull_request:
+    branches: [ "master" ]
+  schedule:
+    - cron: '39 9 * * 0'
```
[the remark this answers is not in the document]
Comment (kevinszuchet) [reply .github/workflows/codeql.yml:7]: My bad. I forgot to change the default.

WDYT about only running this workflow on PRs to master?

[the remark this answers is not in the document]
Comment (cazala) [reply .github/workflows/codeql.yml:7]: yep sounds good to me, we should also do the same on the marketplace right?

[the remark this answers is not in the document]
Comment (kevinszuchet) [reply .github/workflows/codeql.yml:7]: I agree!

**Review by cazala — APPROVED** [review]

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
```
Comment (cyaiox) [inline .github/workflows/codeql.yml:19]: do we want to run multiple language checks in parallel?

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
```
Comment (cyaiox) [inline .github/workflows/codeql.yml:28]: related to the previous question, using `javascript` here works

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (kevinszuchet) [reply .github/workflows/codeql.yml:19]: This comes from the default workflow template. For now, I think we won't check any other language, and, to be honest, I don't have a strong opinion about keeping this here in line 28.

Comment (cyaiox) [reply .github/workflows/codeql.yml:19]: np: do you mind removing this matrix declaration?

**Review by cyaiox — APPROVED** [review]

**Review by LautaroPetaccio — APPROVED** [review]
Great job! 💯

**Pull request merged**